client/txn.go

```diff
@@ -666,7 +666,12 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	if elideEndTxn && pErr == nil {
 		// Check that read only transactions do not violate their deadline.
 		if endTxnRequest.Deadline != nil {
-			if endTxnRequest.Deadline.Less(txn.Proto.Timestamp) {
+			t := txn.Proto.Timestamp
```

Comment:
`BatchRequest.SetActiveTimestamp` doesn't look at `txn.Timestamp` at all, just `txn.OrigTimestamp`. `isEndTransactionExceedingDeadline` only looks at `txn.Timestamp`. I think in this case we want to follow `isEndTransactionExceedingDeadline`, so the code was correct before.